### ISSUE TITLE
Jupyter markdown cells: newcommand and latex formulas

### ISSUE THREAD

**Issue by slel** [sagemathinc/cocalc#4569]
Small trouble with LaTeX formulas with bits involving `\newcommand`
in Jupyter markdown cells.

To reproduce, define two markdown cells as follows in a Jupyter sheet,
the first one not using `\newcommand`, the other one using it.

```
**Cell A**

We denote by $\operatorname{span}(u, v)$ the span of two vectors $u$ and $v$.
```

```
**Cell B**
$\newcommand{\span}{\operatorname{span}}$

We denote by $\span(u, v)$ the span of two vectors $u$ and $v$.
```

Expected: the cells behave exactly the same.

Observed:

1.  The formula is text in **Cell A** but an image in **Cell B**.
2.  When viewing this Jupyter notebook in one's CoCalc project,
    - in **Cell A**
      - one can select text that includes the formula or part of it
      - this selection can be done character by character
      - the selected text gets highlighted in inverted colors (white on blue)
      - the highlighting applies to the selected part of the formula as well
    - in **Cell B**
      - selecting some text, one can include the whole formula or none of it
      - no way to select part of it, character by character
      - highlighting in inverted colors does not apply to the formula if selected
      - if "denote by span(u,v) the span" is selected, "span(u,v)" stays unhighlighted
3.  If this Jupyter notebook is made public on the **CoCalc share server**
    - in **Cell A**
      - the formula "span(u,v)" **gets rendered**
    - in **Cell B**
      - the formula "span(u,v)" **is missing**

**Comment by williamstein** [comment]
newcommand isn't (or wasn't) fully supported by KaTeX when we added support for KaTeX.  When KaTeX fails CoCalc falls back to use MathJax, which does support newcommand.    MathJax is a completely different tex formula renderer, so can have different behavior in subtle ways, e.g., "one can select text that includes the formula or part of it" versus "selecting some text, one can include the whole formula or none of it"...

I pasted
```
\newcommand{\span}{\operatorname{span}}\span(u,v)
```
into the live demo box of the latest version of katex here: https://katex.org/

and it seems to work fine.  Thus if we upgrade katex and configure it properly and it is stateful enough to remember newcommands between different invocation (?), then that will fix this issue.
